src/crimson/osd/pg.cc

```diff
@@ -1049,6 +1055,52 @@ PG::interruptible_future<> PG::handle_rep_op(Ref<MOSDRepOp> req)
     });
 }
 
+void PG::log_operation(
+  std::vector<pg_log_entry_t>&& logv,
+  const eversion_t &trim_to,
+  const eversion_t &roll_forward_to,
+  const eversion_t &min_last_complete_ondisk,
+  bool transaction_applied,
+  ObjectStore::Transaction &txn,
+  bool async) {
+  logger().debug("{}", __func__);
+  if (is_primary()) {
+    ceph_assert(trim_to <= peering_state.get_last_update_ondisk());
+  }
+  if (transaction_applied) {
+    //TODO:
+    //update_snap_map(logv, t);
+  }
+  //auto last = logv.rbegin();
+  //if (is_primary() && last != logv.rend()) {
+    //projected_log.skip_can_rollback_to_to_head();
+    //projected_log.trim(cct, last->version, nullptr, nullptr, nullptr);
+  //}
```

Comment:
?

Reply:
No `projected_log` concept in Crimson's pg yet. It might be easier to track down the missing pieces when grepping these comments while implementing (in similarity to `BackfillState::Enqueuing::maybe_update_range()`).

Reply:
Moved into a `#if 0` block